ec2/elasticloadbalancer.py

```
@@ -0,0 +1,295 @@
+    elb_client = connection.ELBConnectionClient().client()
+
+    try:
+        lb = elb_client.create_load_balancer(**params_dict)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer not created '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('Load Balancer Created. ')
+
+    utils.set_external_resource_id(params_dict['name'],
+                                   ctx.instance,
+                                   False)
+
+    ctx.instance.runtime_properties['elb_name'] = params_dict['name']
+
+    if not lb:
+        raise NonRecoverableError('Load Balancer not created')
+
+    return lb
+
+
+def _get_instance_list():
+
+    ctx.logger.info('Attempting to get Load Balancer Instance List.')
+
+    elb_name = ctx.node.properties['elb_name']
+    lb = _get_existing_elb(elb_name)
+    list_of_instances = lb.instances
+    ctx.instance.runtime_properties[constants.EXTERNAL_RESOURCE_ID] = lb.name
+
+    return list_of_instances
+
+
+def _get_elbs_by_names(list_of_names):
+
+    ctx.logger.info(
+        'Attempting to get Load Balancers {0}.'.format(list_of_names))
+    elb_client = connection.ELBConnectionClient().client()
+
+    try:
+        elb_list = elb_client.get_all_load_balancers(
+            load_balancer_names=list_of_names)
+        total_elb_list = elb_client.get_all_load_balancers()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Error when accessing ELB interface '
+                                  '{0}'.format(str(e)))
+
+    if not elb_list:
+        ctx.logger.error('Unable to find load balancers matching: '
+                         '{0}'.format(list_of_names))
+        ctx.logger.error('load balancers available: '
```

If I remember correctly @EarthmanT requested this for "failing fast" on bootstrap in case the elb was not defined correctly for an existing elb.